src/eureka/S4cal_StellarSpectra/s4cal_StellarSpec.py

```diff
@@ -0,0 +1,285 @@
+        std_t23 = np.ma.std(optspec[it2:it3+1], axis=0)
+        # Mask outliers along wavelength axis
+        tck = spi.splrep(wave, spec_t23, w=1/std_t23, k=3)
+        spline = spi.splev(wave, tck)
+        mask = sigrej(spec_t23 - spline, meta.sigma_thresh)
+        igood = np.where(~mask)[0]
+        # Create XArray data arrays
+        ecl_flux = xrio.makeLCDA(spec_t23[igood, np.newaxis],
+                                 wave[igood], [t0],
+                                 spec.optspec.flux_units,
+                                 spec.optspec.wave_units,
+                                 spec.optspec.time_units,
+                                 name='Median In-Eclipse Flux')
+        ecl_fstd = xrio.makeLCDA(std_t23[igood, np.newaxis],
+                                 wave[igood], [t0],
+                                 spec.optspec.flux_units,
+                                 spec.optspec.wave_units,
+                                 spec.optspec.time_units,
+                                 name='Std. Dev. of In-Eclipse Flux')
+
+        # Create XArray dataset
+        ds = dict(base_flux=base_flux, base_fstd=base_fstd,
+                  ecl_flux=ecl_flux, ecl_fstd=ecl_fstd)
+        ds = xrio.makeDataset(ds)
+        batch.append(ds)
+
+    # Concatenate datasets
+    ds = xrio.concat(batch, dim="time")
+    filename = meta.outputdir+'S4cal_'+meta.eventlabel + "_CalStellarSpec.h5"
+    xrio.writeXR(filename, ds, verbose=True)
+
+    if meta.isplots_S4cal >= 2:
+        plot_stellarSpec(meta, ds)
+
+    return meta, spec, ds
+
+
+def plot_whitelc(optspec, time, meta, i, fig=None, ax=None):
+    '''Plot binned white light curve and indicate
+    baseline and in-eclipse regions.
+    '''
+    toffset = meta.time_offset
+    it0, it1, it2, it3, it4, it5 = meta.it
+
+    # Created binned white LC
+    lc = np.ma.sum(optspec, axis=1)
+    lc /= np.mean(lc)
+    lc_bin = util.binData_time(lc, time, nbin=meta.nbin_plot)
+    time_bin = util.binData_time(time, time, nbin=meta.nbin_plot)
+
+    if i == 0:
+        fig = plt.figure(4202, figsize=(8, 5))
+        plt.clf()
+        ax = fig.subplots(1, 1)
+        ax.plot(time_bin-toffset, lc_bin, '.', color='0.2', alpha=0.8,
+                label='Binned White LC')
+    ymin, ymax = ax.get_ylim()
+    ax.fill_betweenx((ymin, ymax), time[it0]-toffset, time[it1]-toffset,
+                     color=colors[1], alpha=0.2)
+    ax.fill_betweenx((ymin, ymax), time[it4]-toffset, time[it5]-toffset,
+                     color=colors[1], alpha=0.2)
+    ax.fill_betweenx((ymin, ymax), time[it2]-toffset, time[it3]-toffset,
+                     color=colors[0], alpha=0.2)
+    ax.vlines([time[it1]-toffset, time[it4]-toffset,
+              time[it0]-toffset, time[it5]-toffset],
+              ymin, ymax, color=colors[1], label='Baseline Regions')
+    ax.vlines([time[it2]-toffset, time[it3]-toffset],
+              ymin, ymax, color=colors[0], label='In-Eclipse Region')
```

Comment:
Would "In-Occultation" be better, given that this may be applied to transit data as well?